### PR TITLE
Fixes up SCP-2398

### PR DESCRIPTION
**Pull request by HaultyAnonie**
Makes 173 and the like no longer hittable by SCP-2398.

## About the Pull Request

This pull request fixes up SCP-2398 that was previously.. ahem... _coded by the number 7._ - It stops you from hitting non-carbon lifeforms!

## Why It's Good For The Game

173 is no longer hittable, and the likes!

## Changelog

:cl:
Tweak: changed the user [G] to the target [H]
/:cl:

### CODE REVIEW

**Comment by ghost** [comment]
wtf i like slapping 173
he must be hittable for realism even if it doesnt do shit plssssssssss

**Comment by HaultyAnonie** [comment]
> wtf i like slapping 173 he must be hittable for realism even if it doesnt do shit plssssssssss

You can hit it, just not with the bat.

**Comment by TheDarkElites** [comment]
This is poorly done. Not code wise but rather mechanic wise.

First and foremost you should not be prevented from swinging the bat at all, its immersion breaking. Instead the bat should be able to be swung but just have like no effect upon striking 173.

Second this shouldn't check for human, but rather if something has the SCP datum. This stops you from attacking animals or any other non human instead of stopping you from messing with scps specifically.

Thirdly, and this is kind of a code critique, but instead of adding a new check you just replace the check for the user being human??? What??? That check shouldn't be removed if it was there beforehand without good reason.

Finally im refactoring all scp code and this might not be an issue after im done with it.

**Comment by HaultyAnonie** [comment]
> First and foremost you should not be prevented from swinging the bat at all, its immersion breaking. Instead the bat should be able to be swung but just have like no effect upon striking 173.

I was working on a solution for this by using iftypeon the attack proc.

> Second this shouldn't check for human, but rather if something has the SCP datum. This stops you from attacking animals or 
 any other non human instead of stopping you from messing with scps specifically.

> Thirdly, and this is kind of a code critique, but instead of adding a new check you just replace the check for the user being 
 human??? What??? That check shouldn't be removed if it was there beforehand without good reason.
 
 Seven's code...

**Comment by TheDarkElites** [comment]
> Seven's code...

What? No its not, you replaced seven's human check of the attacker with a human check of the attacked. You did that, not seven. You are checking the human attacked, not seven.

File: code/modules/SCP/SCPs/SCP-2398.dm [hunk 1]
```diff
@@ -28,7 +28,7 @@
 			admin_attack_log(G, H, null, null, "[G] has attacked [H] with SCP-2398!")
 			message_staff("[G] (ckey: [G.ckey]) has swung SCP-2398 at [H] ([H.ckey])!") //"im sure theres a proc to format this for admins already; go find it" no.
 	else
-		if(iscarbon(G))
+		if(iscarbon(H))
```
Comment (Tsurupeta) [inline code/modules/SCP/SCPs/SCP-2398.dm]: Replace H with T and G with U

Comment (ghost) [inline code/modules/SCP/SCPs/SCP-2398.dm]: remove all the code

Comment (ghost) [reply code/modules/SCP/SCPs/SCP-2398.dm]: it will get refactored

**Comment by HaultyAnonie** [comment]
> > Seven's code...
>
> What? No its not, you replaced seven's human check of the attacker with a human check of the attacked. You did that, not seven. You are checking the human attacked, not seven.

Funnily enough, he did, he said to change that with H to fix it- I had a shittier way of fixing this.

File: code/modules/SCP/SCPs/SCP-2398.dm [hunk 2]
```diff
@@ -13,29 +13,29 @@
...
 /obj/item/weapon/twohanded/scp_2398/attack(mob/living/target,	mob/living/user)
-	var/mob/living/carbon/human/H = target
-	var/mob/living/carbon/human/G = user
+	var/mob/living/carbon/human/T = target
+	var/mob/living/carbon/human/U = user
 	var/activehand = BP_L_ARM //determine which hand to burn (stolen from 113 code because bay code is messy)
-	if(!G.hand)
+	if(!U.hand)
 		activehand = BP_R_ARM
 	if(ishuman(user && target))
```
Comment (Tsurupeta) [inline code/modules/SCP/SCPs/SCP-2398.dm]: this code never works, replace with individual ishuman checks

**Review by Tsurupeta — CHANGES_REQUESTED** [review]

File: code/modules/SCP/SCPs/SCP-2398.dm (hunk 2 above)
Comment (HaultyAnonie) [reply code/modules/SCP/SCPs/SCP-2398.dm]: I think I did it, please guide me.
👍 Thanks to Elite.

Comment (Tsurupeta) [reply code/modules/SCP/SCPs/SCP-2398.dm]: You broke everything

Comment (TheDarkElites) [reply code/modules/SCP/SCPs/SCP-2398.dm]: What did he break?

File: code/modules/SCP/SCPs/SCP-2398.dm [hunk 3]
```diff
@@ -13,29 +13,19 @@
...
-			to_chat(H, SPAN_USERDANGER("Someone begins swinging a bat at you!"))
-			G.visible_message(SPAN_DANGER("[G] begins to swing [src] at [H]!"))
-			if(do_after(G, 4 SECONDS, H))
-				var/obj/item/organ/external/E = G.get_organ(activehand)
-				E.fracture()
-				explosion(H, 1, 1, 3, 3, 1)
-				admin_attack_log(G, H, null, null, "[G] has attacked [H] with SCP-2398!")
-				message_staff("[G] (ckey: [G.ckey]) has swung SCP-2398 at [H] ([H.ckey])!")
-				qdel(H) //perhaps a bad solution but i dont want to work with it
+	if(isSCP(target) || ishuman(user))
```
Comment (TheDarkElites) [inline code/modules/SCP/SCPs/SCP-2398.dm]:
```suggestion
	if(isSCP(target) || !ishuman(user))
```

File: code/modules/SCP/SCPs/SCP-2398.dm (hunk 2 above)
Comment (TheDarkElites) [reply code/modules/SCP/SCPs/SCP-2398.dm]: he forgot an exclamation mark lmao

File: code/modules/SCP/SCPs/SCP-2398.dm (hunk 3 above)
Comment (HaultyAnonie) [reply code/modules/SCP/SCPs/SCP-2398.dm]: AHHHHH!!! _I didn't have time to double-check the code- but I knew it wasn't working my bad!!!_

File: code/modules/SCP/SCPs/SCP-2398.dm [hunk 4]
```diff
@@ -13,29 +13,19 @@
...
+	if(isSCP(target) || !ishuman(user))
 		return
+	to_chat(T, SPAN_USERDANGER("Someone begins swinging a bat at you!"))
+	U.visible_message(SPAN_DANGER("[U] begins to swing [src] at [T]!"))
+	if(do_after(U, 4 SECONDS, T))
+		var/obj/item/organ/external/E = U.get_organ(activehand)
+		E.fracture()
+		explosion(T, 1, 1, 3, 3, 1)
+		admin_attack_log(U, T, null, null, "[U] has attacked [T] with SCP-2398!")
+		message_staff("[U] (ckey: [U.ckey]) has swung SCP-2398 at [T] ([T.ckey])!") //"im sure theres a proc to format this for admins already; go find it" no.
```
Comment (Tsurupeta) [inline code/modules/SCP/SCPs/SCP-2398.dm:28]: ""im sure theres a proc to format this for admins already; go find it" no." yes

Comment (Tsurupeta) [reply code/modules/SCP/SCPs/SCP-2398.dm:28]: Ok actually no, lol, we need to port this shit, fine

Comment (TheDarkElites) [reply code/modules/SCP/SCPs/SCP-2398.dm:28]: lmao, this was actually what seven wrote. I told him to try to find a proc to format it but @MACIEKBAKI still passed it. Who knew we never even had the proc.

File: code/modules/SCP/SCPs/SCP-2398.dm [hunk 5]
```diff
@@ -13,29 +13,19 @@
 	SCP = /datum/scp/scp_2398
 
 /obj/item/weapon/twohanded/scp_2398/attack(mob/living/target,	mob/living/user)
```
Comment (TheDarkElites) [inline code/modules/SCP/SCPs/SCP-2398.dm]:
```suggestion
/obj/item/weapon/twohanded/scp_2398/attack(mob/living/target,	mob/living/carbon/human/user)
```

File: code/modules/SCP/SCPs/SCP-2398.dm [hunk 6]
```diff
@@ -13,29 +13,19 @@
...
-	var/mob/living/carbon/human/H = target
-	var/mob/living/carbon/human/G = user
+	var/mob/living/carbon/human/T = target
```
Comment (TheDarkElites) [inline code/modules/SCP/SCPs/SCP-2398.dm:16]:
```suggestion

```

File: code/modules/SCP/SCPs/SCP-2398.dm [hunk 7]
```diff
@@ -13,29 +13,19 @@
...
+	var/mob/living/carbon/human/U = user
```
Comment (TheDarkElites) [inline code/modules/SCP/SCPs/SCP-2398.dm:17]:
```suggestion

```

File: code/modules/SCP/SCPs/SCP-2398.dm [hunk 8]
```diff
@@ -13,29 +13,19 @@
...
 	var/activehand = BP_L_ARM //determine which hand to burn (stolen from 113 code because bay code is messy)
-	if(!G.hand)
+	if(!U.hand)
 		activehand = BP_R_ARM
```
Comment (TheDarkElites) [inline code/modules/SCP/SCPs/SCP-2398.dm]: replace with var/activehand = user.hand ? BP_L_HAND : BP_R_HAND

File: code/modules/SCP/SCPs/SCP-2398.dm [hunk 9]
```diff
@@ -13,29 +13,19 @@
...
-			to_chat(H, SPAN_USERDANGER("Someone begins swinging a bat at you!"))
-			G.visible_message(SPAN_DANGER("[G] begins to swing [src] at [H]!"))
-			if(do_after(G, 4 SECONDS, H))
-				var/obj/item/organ/external/E = G.get_organ(activehand)
-				E.fracture()
-				explosion(H, 1, 1, 3, 3, 1)
-				admin_attack_log(G, H, null, null, "[G] has attacked [H] with SCP-2398!")
-				message_staff("[G] (ckey: [G.ckey]) has swung SCP-2398 at [H] ([H.ckey])!")
-				qdel(H) //perhaps a bad solution but i dont want to work with it
+	if(isSCP(target) || !ishuman(user))
```
Comment (TheDarkElites) [inline code/modules/SCP/SCPs/SCP-2398.dm]:
```suggestion
	if(isSCP(target) || !istype(user) || !istype(target))
```

**Review by TheDarkElites — CHANGES_REQUESTED** [review]

**Comment by projectkepler-ru** [comment]
> > > Seven's code...
> > 
> > 
> > What? No its not, you replaced seven's human check of the attacker with a human check of the attacked. You did that, not seven. You are checking the human attacked, not seven.
> 
> Funnily enough, he did, he said to change that with H to fix it- I had a shittier way of fixing this.

I don't think you should take seven's answer for granted

**Comment by TheDarkElites** [comment]
refactored in #1205 ([commit here](https://github.com/Foundation-19/Foundation-19/pull/1205/commits/bf38d7492e06f391dfd0b3256de0e34c3e084c6e)) so theres no point in merging this anyway.

**Pull request closed without merging**